client/src/extension.ts

```diff
@@ -34,10 +34,20 @@ export function activate(context: vscode.ExtensionContext): Thenable<any> {
 		return location;
 	}
 
-	function initVerifier(fileUri: URI): (location: Location) => void {
-		return location => {
+	function initVerifier(fileUri: URI): (location: Location) => Promise<void> {
+		return async location => {
 			const fileData = new FileData(fileUri);
-			const verifierConfig = new VerifierConfig(location, [fileData]);
+			const z3Path = Helper.getZ3Path(location);
+			const boogiePath = Helper.getBoogiePath(location);
+			if (z3Path.error != null) {
+				vscode.window.showErrorMessage(z3Path.error);
+				return Promise.reject(z3Path.error);
+			}
+			if (boogiePath.error != null) {
+				vscode.window.showErrorMessage(boogiePath.error);
+				return Promise.reject(boogiePath.error);
+			}
+			const verifierConfig = new VerifierConfig([fileData], z3Path.path, boogiePath.path);
 			Verifier.initialize(context, verifierConfig, fileUri);
 			Notifier.notifyExtensionActivation();
 		}
```

Comment:
why are we making this async?

Reply:
You're right, I can solve it with throwing exceptions (which is equivalent to returning a rejected promise)